File: leetcode/golang/LCR_158_shu_zu_zhong_chu_xian_ci_shu_chao_guo_yi_ban_de_shu_zi_lcof.go
func inventoryManagement(stock []int) int {
  vote, x := 0, 0
  for i := 0; i < len(stock); i++ {
      if vote == 0 {
          x = stock[i]
      }
      if x == stock[i] {
          vote += 1
      } else {
          vote -= 1
      }
  }
  return x
}

File: leetcode/c++/LCR_158_shu_zu_zhong_chu_xian_ci_shu_chao_guo_yi_ban_de_shu_zi_lcof.cpp
// https://leetcode.cn/problems/shu-zu-zhong-chu-xian-ci-shu-chao-guo-yi-ban-de-shu-zi-lcof

#include <bits/stdc++.h>
using  namespace std;

class Solution {
public:
    int inventoryManagement(vector<int>& stock) {
      int vote = 0, x = 0;
      for (auto num : stock) {
        if (vote == 0) {
            x = num;
        }
        vote += num == x? 1 : -1;
      }
      return x;
    }
};

// 摩尔投票法，测试了一下排序然后取中间的数最快